clippy_lints/src/disallowed_macros.rs

```diff
@@ -68,14 +68,10 @@ pub struct DisallowedMacros {
     // Track the most recently seen node that can have a `derive` attribute.
     // Needed to use the correct lint level.
     derive_src: Option<OwnerId>,
-
-    // When a macro is disallowed in an early pass, it's stored
-    // and emitted during the late pass. This happens for attributes.
-    early_macro_cache: AttrStorage,
```

Comment:
It is not clear to me what the plan is for `AttrStorage` and its usage in `DeprecatedMacros` and elsewhere.  Since it was introduced early in the attribute refactor, perhaps it was a temporary solution.

@jdonszelmann Could you please take a look?